frontend/src/pages/offers/ViewOfferPage.tsx

```diff
@@ -59,7 +78,37 @@ const ViewOfferPage: FunctionComponent = () => {
       offerId,
     }
 
-    addPallet({ variables: { input: newPallet } }).then(() => [hideModal()])
+    addPallet({ variables: { input: newPallet } }).then(hideCreateModal)
+  }
+
+  const [
+    deleteModalIsVisible,
+    showDeleteModal,
+    hideDeleteModal,
+  ] = useModalState()
+
+  const [selectedPallet, setSelectedPallet] = useState<number>()
+
+  const selectPalletToDestroy = (palletId: number) => {
+    setSelectedPallet(palletId)
+    showDeleteModal()
+  }
+
+  // The mutation below will automatically update the cache because the API
+  // returns the offer. However, it'll create a warning in the console, and I
+  // can't find a way to get rid of it without some redundant configuration
+  // "Cache data may be lost when replacing the pallets field of a Offer object"
```

Comment:
This is reaaaaaaaaally annoying.

<img width="572" alt="Screen Shot 2021-03-25 at 12 19 28 PM" src="https://user-images.githubusercontent.com/3411183/112531075-62df9d00-8d64-11eb-97c0-213bd15fe6d3.png">

Based on the error message and the StackOverflow issues I've read (the docs weren't helpful), Apollo tries to merge the `existing` and `incoming` values of the `pallets` array after a deletion. All that's needed is for the `incoming` array to replace what's already there, so it's doing the right thing, but I can't find a way to disable this warning besides added a custom `merge` function to the resolver, which is redundant because IT'S DOING THE RIGHT THING ugh.

Reply:
Yeah, that sucks. Since the merge function would be super quick to write how do you feel about adding it?

Reply:
I hate it but I did it